### PR TITLE
Update release checklist, post 0.78.0

### PR DESCRIPTION
**Pull request by christophermaier**
Various process updates following the 0.78.0 release.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -34,7 +36,6 @@ Here are all the steps for the release process. Create a new issue at the beginn
 - [ ] [Promote Windows builder worker](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#promote-the-builder-worker) and confirm build uses new version
 - [ ] Post announcement in [Chef discourse](https://discourse.chef.io/c/habitat)
 - [ ] Post announcement in [Habitat forums](https://forums.habitat.sh/c/announcements)
-- [ ] Tweet announcement from `@habitatsh`
```
Comment (mwrock) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: We should also delete the line in `release.md` that says to do this.

Comment (christophermaier) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: @mwrock Good catch; fixed in https://github.com/habitat-sh/habitat/pull/6307/commits/a9741bc2ecda2514362967c1c1a3d66368ef0825

**Comment by christophermaier** [comment]
See habitat-sh/habitat#6284

**Pull request merged**